### DISCUSSION TITLE
erro ao executar curso:  Criando Apps Web com Angular e Redux

### DISCUSSION THREAD

**Discussion by EmerApOli** [balta-io/forum#151] [Tira Dúvidas] [answered]
Error: src/app/app.component.ts:14:36 - error TS2769: No overload matches this call.
[ng]   Overload 1 of 9, '(mapFn: (state: CartModel) => CartModel): (source$: Observable<CartModel>) => Observable<CartModel>', gave the following error.
[ng]     Argument of type 'string' is not assignable to parameter of type '(state: CartModel) => CartModel'.
[ng]   Overload 2 of 9, '(key: keyof CartModel): (source$: Observable<CartModel>) => Observable<number | ProductModel[]>', gave the following error.
[ng]     Argument of type '"cart"' is not assignable to parameter of type 'keyof CartModel'.
[ng]
[ng] 14     this.cart$ = store.pipe(select('cart'));
[ng]                                       ~~~~~~
[ng]
[ng]
[ng]
[ng] Error: src/app/app.module.ts:27:25 - error TS2322: Type '(state: CartModel | undefined, action: ActionModel) => CartModel' is not assignable to type 'ActionReducer<CartModel, Action>'.
[ng]   Types of parameters 'action' and 'action' are incompatible.
[ng]     Property 'payload' is missing in type 'Action' but required in type 'ActionModel'.
[ng]
[ng] 27    StoreModule.forRoot({cart: cartReducer}),
[ng]                            ~~~~
[ng]
[ng]   src/app/models/action.model.ts:5:5
[ng]     5     payload: any;
[ng]           ~~~~~~~
[ng]     'payload' is declared here.
[ng]
[ng]ja removi o projeto deixei com as mesmas versçoes e nada

[ng]
[ng] × Failed to compile.ucer}),
]ja removi o projeto deixei com as mesmas versçoes e nada

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @EmerApOli, como vai?
As versões do angular e pacotes estão iguais as do curso? Precisam ser as mesmas...
